partition/roles/metal-core/tasks/main.yaml

```
@@ -14,6 +14,10 @@
       - metal_core_image_tag is defined
       - metal_core_rack_id is not none
 
+- name: Set service dependency
+  set_fact:
+    metal_core_service_dependency: "{{ database.service if metal_stack_switch_os_is_sonic else docker.service }}"
```

I think this has to be quoted, otherwise in the Jinja context it will be interpreted as a variable:

```suggestion
    metal_core_service_dependency: "{{ 'database.service' if metal_stack_switch_os_is_sonic else 'docker.service' }}"
```